analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -571,15 +571,36 @@ open class Npm(
     /**
      * Install dependencies using the given package manager command.
      */
-    private fun installDependencies(workingDir: File) {
+    private fun installDependencies(workingDir: File): List<OrtIssue> {
         requireLockfile(workingDir) { hasLockFile(workingDir) }
 
         // Install all NPM dependencies to enable NPM to list dependencies.
         val process = runInstall(workingDir)
 
-        // TODO: Capture warnings from npm output, e.g. "Unsupported platform" which happens for fsevents on all
-        //       platforms except for Mac.
-        process.stderr.withoutPrefix("Error: ")?.also { throw IOException(it.lineSequence().first()) }
+        val lines = process.stderr.lines()
+        val issues = mutableListOf<OrtIssue>()
+        val commonSecondaryPrefixes = listOf("deprecated ", "enoent ", "old lockfile ")
+
+        fun mapLinesToOrtIssues(prefix: String, severity: Severity) {
+            val issueLines = lines.takeWhile { it.startsWith(prefix) }.mapNotNull { line ->
+                line.removePrefix(prefix).let {
+                    commonSecondaryPrefixes.fold(it) { remainder, prefix -> remainder.removePrefix(prefix) }
+                }.takeUnless { it.isBlank() }
+            }
+
+            if (issueLines.isNotEmpty()) {
+                issues += OrtIssue(
+                    source = managerName,
+                    message = issueLines.joinToString("\n"),
+                    severity = severity
+                )
+            }
+        }
+
+        mapLinesToOrtIssues("npm WARN ", Severity.WARNING)
+        mapLinesToOrtIssues("npm ERR! ", Severity.ERROR)
```

Comment:
I would like to have a test case which also tests the error path, I would be fine with adding that later.

Reply:
This case seems to be a bit difficult to trigger consistently across different NPM versions, so I'd indeed prefer to add this later.

Reply:
I found a way to trigger that error and added a test.

Reply:
Did you add that already? In the expected results diffs I only see warnings.

Reply:
> Did you add that already?

Yes.

> In the expected results diffs I only see warnings.

I didn't bother to create / modify an external result file, but added a new test that tests this programmatically:

https://github.com/oss-review-toolkit/ort/pull/6243/files#diff-27797998f83fc67dfd200ad5f84575223bca6e55450685f75ca392c3dd0f9753R111-R123

Reply:
Thanks, I missed that.